core/application/bloodDonationWorkflow/DonorSearchService.ts

```diff
@@ -0,0 +1,180 @@
+    sourceQueueArn: string,
+    userProfile: UserDetailsDTO,
+    donorSearchRepository: Repository<DonorSearchDTO>,
+    stepFunctionModel: StepFunctionModel
+  ): Promise<void> {
+    const { seekerId, requestPostId, createdAt } = donorRoutingAttributes
+
+    const donorSearchRecord = await donorSearchRepository.getItem(
+      `${DONOR_SEARCH_PK_PREFIX}#${seekerId}`,
+      `${DONOR_SEARCH_PK_PREFIX}#${createdAt}#${requestPostId}`
+    )
+
+    if (donorSearchRecord === null) {
+      await donorSearchRepository.create({
+        id: donorRoutingAttributes.requestPostId,
+        ...donorRoutingAttributes,
+        status: DonationStatus.PENDING,
+        retryCount: 0
+      })
+    } else if (donorSearchRecord.status === DonationStatus.COMPLETED) {
+      if (
+        sourceQueueArn === process.env.DONOR_SEARCH_QUEUE_ARN &&
+        donorRoutingAttributes.bloodQuantity > donorSearchRecord.bloodQuantity
+      ) {
+        const updatedRecord: Partial<DonorSearchDTO> = {
+          ...donorRoutingAttributes,
+          id: requestPostId,
+          status: DonationStatus.PENDING,
+          retryCount: 0
+        }
+        await donorSearchRepository.update(updatedRecord)
+      }
+    }
+
+    const retryCount = donorSearchRecord?.retryCount ?? 0
+    const updatedRecord: Partial<DonorSearchDTO> = {
+      ...donorRoutingAttributes,
+      id: requestPostId,
+      retryCount: retryCount + 1
+    }
+
+    if (retryCount >= Number(process.env.MAX_RETRY_COUNT)) {
+      updatedRecord.status = DonationStatus.COMPLETED
+      await donorSearchRepository.update(updatedRecord)
+      return
+    }
+
+    await donorSearchRepository.update(updatedRecord)
+
+    const stepFunctionPayload: StepFunctionInput = {
+      seekerId,
+      requestPostId,
+      createdAt,
+      donationDateTime: donorRoutingAttributes.donationDateTime,
+      requestedBloodGroup: donorRoutingAttributes.requestedBloodGroup,
+      bloodQuantity: Number(donorRoutingAttributes.bloodQuantity),
+      urgencyLevel: donorRoutingAttributes.urgencyLevel,
+      geohash: donorRoutingAttributes.geohash,
+      seekerName: userProfile.name,
+      patientName: donorRoutingAttributes.patientName ?? '',
+      location: donorRoutingAttributes.location,
+      contactNumber: donorRoutingAttributes.contactNumber,
+      transportationInfo: donorRoutingAttributes.transportationInfo ?? '',
+      shortDescription: donorRoutingAttributes.shortDescription ?? '',
+      city: donorRoutingAttributes.city,
+      retryCount: retryCount + 1,
+      message: getBloodRequestMessage(
+        donorRoutingAttributes.urgencyLevel,
+        donorRoutingAttributes.requestedBloodGroup,
+        donorRoutingAttributes.shortDescription ?? ''
+      )
+    }
+
+    await stepFunctionModel.startExecution(
+      stepFunctionPayload,
+      `${requestPostId}-${donorRoutingAttributes.city}-(${donorRoutingAttributes.requestedBloodGroup})-${Math.floor(Date.now() / 1000)}`
+    )
+    return
```

Comment:
no need for empty return

Reply:
i need to stop the function here

Reply:
the function will stop here irregardless. Given it doesn't have any more code.

Reply:
Updated in 41692a62c95870d78ffb56fb476a6d84c7a30dc1